distributed/core.py

```diff
@@ -83,6 +83,11 @@ def _raise(*args, **kwargs):
 LOG_PDB = dask.config.get("distributed.admin.pdb-on-err")
 
 
+# Cached version of is_coroutine_function for server stream handlers,
+# allowing for quicker checks on on frequent code paths.
+_is_coroutine_function = lru_cache(1000)(is_coroutine_function)
```

Comment:
Should we just decorate [the definition]( https://github.com/dask/distributed/blob/3407aa31be08ce100bd96e42acb99e1d7daf996e/distributed/utils.py#L1346 )?

Also how did we conclude 1,000 is the right size here?

Reply:
> Should we just decorate [the definition](https://github.com/dask/distributed/blob/3407aa31be08ce100bd96e42acb99e1d7daf996e/distributed/utils.py#L1346)?

Sure, I'm happy to do that as well, I just usually try to default to minimal API changes (though this seems pretty harmless) :)

> Also how did we conclude 1,000 is the right size here?

No reason, really. Seemed unlikely to me that more than those would be checked, especially in this context. Could also be `None`.